sui.py

```diff
@@ -0,0 +1,203 @@
+    """
+    try:
+        payload = {
+            "jsonrpc": "2.0",
+            "id": 1,
+            "method": "suix_getCoinMetadata",
+            "params": [coin_type],
+        }
+
+        response = http_json_call(
+            url=rpc_url,
+            rpc_call_status_counter=rpc_call_status_counter,
+            params=payload,
+            method="POST",
+        )
+
+        if "result" in response and response["result"]:
+            result = response["result"]
+            return {
+                "decimals": result.get("decimals", 0),
+                "symbol": result.get("symbol", "UNKNOWN"),
+                "name": result.get("name", ""),
+            }
+        else:
+            # Default fallback
+            return {"decimals": 0, "symbol": "UNKNOWN", "name": ""}
+
+    except Exception:
+        # If metadata fetch fails, return defaults
+        return {"decimals": 0, "symbol": "UNKNOWN", "name": ""}
+
+
+def get_sui_coin_balance_by_symbols(
+    rpc_url, address, symbols=None, rpc_call_status_counter=None
+):
+    """
+    Get balance for specific coins by symbols.
+    Always includes SUI balance, plus any additional symbols requested.
+
+    Args:
+        rpc_url: The Sui RPC endpoint URL
+        address: The Sui wallet address
+        symbols: Optional comma-separated string or list of symbols
+                 (e.g., 'IKA' or 'IKA,WAL' or ['IKA', 'WAL']).
+                 If None/empty, only returns SUI balance.
+        rpc_call_status_counter: Prometheus counter for RPC call status
+
+    Returns:
+        list: List of dicts with 'balance' and 'symbol' keys
+    """
+    try:
+        # Parse symbols parameter
+        target_symbols = []
+        if symbols:
+            if isinstance(symbols, str):
+                target_symbols = [s.strip().upper() for s in symbols.split(",")]
+            elif isinstance(symbols, list):
+                target_symbols = [s.strip().upper() for s in symbols]
+
+        # Get all coin balances for the address
+        payload = {
+            "jsonrpc": "2.0",
+            "id": 1,
+            "method": "suix_getAllBalances",
+            "params": [address],
+        }
+
+        response = http_json_call(
+            url=rpc_url,
+            rpc_call_status_counter=rpc_call_status_counter,
+            params=payload,
+            method="POST",
+        )
+
+        if "result" not in response:
+            raise Exception(f"Invalid response from Sui RPC: {response}")
+
+        balances = []
+        sui_found = False
+        found_symbols = set()
+
+        # Process all coin types
+        for coin_balance in response["result"]:
+            coin_type = coin_balance.get("coinType", "")
+            total_balance = int(coin_balance.get("totalBalance", 0))
+
+            # Always include SUI (gas token)
+            if coin_type == "0x2::sui::SUI":
+                # Fetch metadata to get decimals for SUI
+                metadata = get_coin_metadata(
```

Comment:
let's skip this RPC call to reduce the number of RPC request. SUI COIN Type should always have a decimal of 9